src/Compilers/CSharp/Portable/Symbols/SignatureOnlyMethodSymbol.cs

```diff
@@ -35,13 +36,15 @@ internal sealed class SignatureOnlyMethodSymbol : MethodSymbol
             ImmutableArray<TypeParameterSymbol> typeParameters,
             ImmutableArray<ParameterSymbol> parameters,
             RefKind refKind,
+            bool isInitOnly,
             TypeWithAnnotations returnType,
             ImmutableArray<CustomModifier> refCustomModifiers,
             ImmutableArray<MethodSymbol> explicitInterfaceImplementations)
         {
             _callingConvention = callingConvention;
             _typeParameters = typeParameters;
             _refKind = refKind;
+            _isInitOnly = isInitOnly;
```

Comment:
>isInitOnly [](start = 26, length = 10)

Could we add an assert that if this is true then we have required modifier in returnType?

Reply:
I don't think that's useful

---
In reply to: [415130732](https://github.com/dotnet/roslyn/pull/43275#discussion_r415130732) [](ancestors = 415130732)

Reply:
> I don't think that's useful

I think it is useful to understand if that condition is violated and when. Because sometimes we are comparing with paying attention to modifiers, so we want to be careful here. I am not suggesting that we have to remove the ```_isInitOnly``` field. Just want to make sure we always have consistent state, as we understand it 

---
In reply to: [415236027](https://github.com/dotnet/roslyn/pull/43275#discussion_r415236027) [](ancestors = 415236027,415130732)